roles/map-role/tasks/main.yml

```diff
@@ -0,0 +1,92 @@
+---
+################################################################################
+# Dynamically-Assigned Role Values (static values are initialized in inventory)
+################################################################################
+# glance_endpoint_address
+- block:
+  - set_fact:
+      glance_endpoint_address: "{{ha_cluster_ip}}"
+
+  # neutron_l3_agent_mode
+  - set_fact:
+      neutron_l3_agent_mode: "dvr_snat"
+    when: dvr == "on" and snat == "on"
+  - set_fact:
+      neutron_l3_agent_mode: "dvr"
+    when: dvr == "on" and snat == "off"
+  - set_fact:
+      neutron_l3_agent_mode: "dvr"
+    when: neutron_l3_agent_mode is undefined
+
+  # neutron_ovs_enable_distributed_routing
+  - set_fact:
+      neutron_ovs_enable_distributed_routing: "True"
+    when: dvr == "on"
+  - set_fact:
+      neutron_ovs_enable_distributed_routing: "False"
+    when: dvr == "off"
+  - set_fact:
+      neutron_ovs_enable_distributed_routing: "True"
+    when: dvr is undefined
+
+  # neutron_novncproxy_base_url
+  - set_fact:
+      neutron_novncproxy_base_url: "http://{{ha_cluster_ip}}:6080/vnc_auto.html"
+    when: neutron_novncproxy_base_url is undefined
+  when: rolename != "pf9-kube"
+
+################################################################################
+# Configure Role Metadata
+################################################################################
+- name: validate pf9/host_id.conf
+  stat:
+    path: /etc/pf9/host_id.conf
+  register: stat_hostid
+
+- fail: msg="failed to open /etc/pf9/host_id.conf"
+  when: stat_hostid.stat.exists == false
+
+- name: get host_id
+  shell: cat /etc/pf9/host_id.conf | grep ^host_id | cut -d = -f2 | cut -d ' ' -f2
```

Comment:
Use the command module.

`command: grep -oP '(?<=host_id = )(.*)' /etc/pf9/host_id.conf`